docs/development/release.md

```diff
@@ -63,7 +66,9 @@ Note: This is for 1.10.3 or later.
 
 ### 6. Create a pull request against the minor version branch
 
-Create a PR against the minor version branch. Verify that the Bamboo build for the PR succeeds and then merge to the minor version branch.
+Push the release branch to GitHub.
+Create a PR against the minor version base branch. Verify that the Bamboo build for the PR succeeds and then merge to the minor version base branch.
```

Comment:
I'm wondering if we should advise against squash-merging these PRs, because it seems like it messes the up the compare view linked in the changelog.